### PR TITLE
Fix some bugs and make usability enhancements

### PR DESCRIPTION
**Pull request by devinjameson**
- Style habit list item correctly when a habit is complete
- Fix an issue with the Habit type
- Update typographic styles
- Store new habit list after habit deletion
- Move add habit button to navigation header
- Update various styles

### CODE REVIEW

File: index.tsx [hunk 1]
```diff
@@ -35,6 +40,10 @@ Navigation.registerComponent(
   "com.buildingHabits.AddHabitScreen",
   () => AddHabitScreen,
 )
+Navigation.registerComponent(
```
Comment (johnschoeman) [inline index.tsx:43]: interesting that you have to register these components, (but no surprising when you think about it as these will live in the native header.)

File: src/Home.tsx [hunk 2]
```diff
@@ -96,17 +73,22 @@ const HabitListItem: FunctionComponent<HabitListItemProps> = ({
       : store.completeHabit(habit.id)
   }
 
-  const daysCompleted = habit.history.length
-  const habitProgressText = `${
-    store.totalNumberOfDays - daysCompleted
-  } days left`
+  const daysCompleted =
+    habit.history.length > store.totalNumberOfDays
```
Comment (johnschoeman) [inline src/Home.tsx:77]: `store.totalNumberOfDays` strikes me as not wanting to be a part of the store, as it's less to do with application state and more to do with business logic.

**Review by johnschoeman — COMMENTED** [review]
Noice

File: src/Home.tsx (hunk 2 above)
Comment (devinjameson) [reply src/Home.tsx:77]: What if we want to enable the user to change the total number of days?

**Pull request merged**